js/store.js

```diff
@@ -134,19 +256,74 @@ const store = {
     const firstParty = this.getWebsite(origin);
     const thirdParty = this.getWebsite(target);
 
+/*
+  if not {},
+
+  firstParty is expected to match this format:
+  {
+    thirdPartyHostnames: [
+      "www.thirdpartydomain.com",
+      ...
+    ]
+  }
+  thirdParty is expected to match this format:
+  {
+    firstPartyHostnames: [
+      "www.firstpartydomain.com",
+      ...
+    ],
+    isIgnored: true, (optional)
+    whiteListedFirstParty: "www.google.com" (optional)
+  }
+  */
+
+    let whiteListOverride = false;
+    let onWhiteList = false;
+
     if (!('thirdPartyHostnames' in firstParty)) {
```

Comment:
Overall I think this code would be much simpler if:

- you store state of isIgnored/isHidden/isVisible or whatever var name you choose
- you only return websites that are visible in `getAll`
- you keep storing firstPartyHostnames in third party as this will never be seen until visible
- when this code decides it becomes visible, it should make sure all websites in `firstPartyHostnames`  have a reference to the thirdParty in their `thirdPartyHostnames` arrays.
- when a website becomes visible the update should fire, the update should not fire for this website until then